internal/pyxis/submit.go

```diff
@@ -134,3 +172,68 @@ func normalizeDockerRegistry(registry string) string {
 
 	return registry
 }
+
+// artifactsTar takes a source path and a writer; a tar writer loops over the files in the source
+// directory, writes the appropriate header information and copies the file into the tar writer
+//
+//nolint:unparam // ctx is unused. Keep for future use.
+func artifactsTar(ctx context.Context, src string, w io.Writer) error {
```

Comment:
I think what you're asking is to use `afero.Fs` here, but I think that would make it more confusing of an abstraction. All of the places in the `engine` that write files use `path` and `src` or `dst` as a string, this follows that pattern.

If we didn't want to follow that pattern, maybe since the caller of this is a `FileSystemWriter` which containrs an afero.FS (though that wouldn't be used in the implementation, just the `dir string`) this method could take the `FileSystemWriter`?

Reply:
That's fair. No need to hold this up. We can revisit another time.